cdap-app-templates/cdap-etl/cdap-etl-batch/src/test/java/co/cask/cdap/etl/batch/ETLBatchRuntimeDatasetTest.java

```diff
@@ -155,4 +156,46 @@ public void testNoMacroMapReduce() throws Exception {
     workflowManager.start();
     workflowManager.waitForFinish(5, TimeUnit.MINUTES);
   }
+
+
+  /**
+   * Tests the secure storage macro function by creating dataset names from the secure store data.
+   */
+  @Test
+  public void testSecureStoreMacroMapReduce() throws Exception {
```

Comment:
lets move this into the DataPipelineTest and have a test case for Spark as well.